modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBMessageLocalServiceImpl.java

```diff
@@ -2511,12 +2539,88 @@ else if (serviceContext.isCommandUpdate() &&
 				messageSubject = messageSubject.substring(
 					messageSubjectPrefix.length());
 			}
+
+			if (FeatureFlagManagerUtil.isEnabled("LPS-182020")) {
+				String newLine = StringPool.NEW_LINE;
+				String quotedLine = StringPool.GREATER_THAN + StringPool.SPACE;
+
+				if (htmlFormat) {
+					newLine = "<br />";
+					quotedLine = StringPool.BLANK;
+				}
+
+				MBMessage rootMessage = mbMessageLocalService.getMessage(
+					message.getRootMessageId());
```

Comment:
Not sure if this is the real root message or an empty message that is created as thread root automatically. We should verify this.

Reply:
Hi, thanks for checking

Yes, this is the rootMessage as far I see.
When you create a new thread, the `rootMessageId` and the `messageId` are the same on the  `MBMessage` table.

Thanks for checking it :)